### PR TITLE
24 Added differentiation between root and non-root tasks

### PR DESCRIPTION
**Pull request by RunCor399**
Tasks in the nifi role have been differentiated in tasks that require root permissions and tasks that doesn't.
This is something that allows to install nifi by leveraging the nifi user, given that all the tasks that require root permissions had already been performed beforehand

### CODE REVIEW

File: handlers/main.yml [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+#   become_user: "{{ nifi_user }}"
+#   systemd:
+#     name: nifi
+#     state: restarted
+#     # scope: user
+#   # environment:
+#   #   XDG_RUNTIME_DIR: "/run/user/{{ myuid_nifi.stdout }}"
+#   tags:
+#   - root
+#   - no-root
```
Comment (lorenzo-romanelli) [inline handlers/main.yml]: I think all commented code can be removed

File: tasks/install.yml [hunk 2]
```diff
@@ -1,44 +1,91 @@
...
+  file:
+    path: "/tmp"
+    state: directory
+    owner: "{{ nifi_user }}"
+    group: "{{ nifi_group }}"
+    recurse: "yes"
+  tags:
+  - root
+
+- name: Fix permissions for systemd # Make sure its safe
```
Comment (lorenzo-romanelli) [inline tasks/install.yml]: is it safe? :)

File: tasks/install.yml [hunk 3]
```diff
@@ -1,44 +1,91 @@
...
     owner: "{{ nifi_user }}"
     group: "{{ nifi_group }}"
     creates: "{{ nifi_config_dirs.install }}/nifi-{{ nifi_version }}"
+  tags:
+  - root
+  - no-root
 
-- name: Fix permissions
+- name: Fix permissions # This may be unnecessary because we are already downloading nifi as the nifi_user
+  become: yes
```
Comment (lorenzo-romanelli) [inline tasks/install.yml]: I would also verify that this is really needed

File: tasks/install.yml [hunk 4]
```diff
@@ -1,44 +1,91 @@
...
+- name: Fix permissions for lineinfile
+  become: yes
+  file:
+    path: "/tmp"
+    state: directory
+    owner: "{{ nifi_user }}"
+    group: "{{ nifi_group }}"
+    recurse: "yes"
+  tags:
+  - root
```
Comment (lorenzo-romanelli) [inline tasks/install.yml]: this seems a bit unsafe, as we are giving the ownership of the entire tmp folder to the nifi user

File: tasks/install.yml (hunk 2 above)
Comment (RunCor399) [reply tasks/install.yml]: I've removed the task as it is not necessary

File: tasks/install.yml (hunk 3 above)
Comment (RunCor399) [reply tasks/install.yml]: Also this one wasn't really necessary for the reason specified in the comment

File: tasks/install.yml (hunk 4 above)
Comment (RunCor399) [reply tasks/install.yml]: Modified to only give ownership on /tmp/.ansible/* to nifi user. The same has also been done in the nifi registry role

**Pull request merged**